service/src/java/org/apache/hive/service/cli/operation/OperationLogManager.java

```diff
@@ -0,0 +1,387 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hive.service.cli.operation;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.QueryInfo;
+import org.apache.hadoop.hive.ql.QueryState;
+import org.apache.hadoop.hive.ql.session.OperationLog;
+import org.apache.hadoop.util.StringUtils;
+import org.apache.hive.service.cli.OperationHandle;
+import org.apache.hive.service.cli.session.HiveSession;
+import org.apache.hive.service.cli.session.HiveSessionImpl;
+import org.apache.hive.service.cli.session.SessionManager;
+
+/**
+ * Move the operation log into another log location that different from the dir created by
+ * {@link HiveSessionImpl#setOperationLogSessionDir(File)},
+ * this will avoid the operation log being cleaned when session/operation is closed, refer to
+ * {@link HiveSessionImpl#close()}, so we can get the operation log for the optimization
+ * and investigating the problem of the operation handily for users or administrators.
+ * The tree under the log location looks like:
+ * - ${@link SessionManager#operationLogRootDir}_historic
+ *    - sessionId
+ *        - queryId (the operation log file)
+ * <p>
+ * while the origin tree would like:
+ * - ${@link SessionManager#operationLogRootDir}
+ *    - sessionId
+ *        - queryId (the operation log file)
+ * <p>
+ * The lifecycle of the log is managed by a daemon called {@link OperationLogDirCleaner},
+ * it gets all query info stored in {@link QueryInfoCache}, searches for the query info that can not be reached on the webui,
+ * and removes the log. If the operation log session directory has no operation log under it and the session is dead,
+ * then the OperationLogDirCleaner will try to cleanup the session log directory.
+ */
+
+public class OperationLogManager {
+  private static final Logger LOG = LoggerFactory.getLogger(OperationLogManager.class);
+  private static final String HISTORIC_DIR_SUFFIX = "_historic";
+  private static String HISTORIC_OPERATION_LOG_ROOT_DIR;
+  private static long MAX_BYTES_TO_FETCH;
```

Comment:
These are not final. I think we usually try to reserve this naming format for `static final` variables

Reply:
done